conclave-common/src/main/kotlin/com/r3/conclave/common/internal/CallInterface.kt

```diff
@@ -33,6 +33,7 @@ abstract class CallInterface<OUTGOING_CALL_TYPE, INCOMING_CALL_TYPE> {
      * Add a call handler for a specific call type.
      * If the call type already has a handler, throw an exception.
      */
+    @Synchronized
```

Comment:
You're using `ConcurrentHashMap`, so just make use of `putIfAbsent` to avoid locking as much as possible.

Reply:
How will the error condition be detected in that case?

Reply:
Have a look at the javadoc - it returns null if there is no previous mapping